src/Umbraco.Web/Strategies/ServerRegistrationEventHandler.cs

```diff
@@ -1,149 +1,95 @@
-                if (e.HttpContext.Request.Url.IsBackOfficeRequest(HttpRuntime.AppDomainAppVirtualPath))
-                {
-                    //yup it's a back office request!
-                    using (var lck = new UpgradeableReadLock(Locker))
-                    {
-                        //we don't want to update if it's not been at least a minute since last time
-                        var isItAMinute = DateTime.Now.Subtract(_lastUpdated).TotalSeconds >= 60;
-                        if (isItAMinute)
-                        {
-                            lck.UpgradeToWriteLock();
-                            _initUpdated = true;
-                            _lastUpdated = DateTime.Now;
-                            UpdateServerEntry(e.HttpContext, e.UmbracoContext.Application);
-                        }
-                    }
-                }
+                case EnsureRoutableOutcome.IsRoutable:
+                    // front-end request
+                    RegisterServer(e);
+                    break;
+                case EnsureRoutableOutcome.NotDocumentRequest:
+                    // anything else (back-end request, service...)
+                    //so it's not a document request, we'll check if it's a back office request
+                    if (e.HttpContext.Request.Url.IsBackOfficeRequest(HttpRuntime.AppDomainAppVirtualPath))
+                        RegisterServer(e);
+                    break;
+                /*
+                case EnsureRoutableOutcome.NotReady:
+                case EnsureRoutableOutcome.NotConfigured:
+                case EnsureRoutableOutcome.NoContent:
+                default:
+                    // otherwise, do nothing
+                    break;
+                */
             }
         }
 
-
-        private static void UpdateServerEntry(HttpContextBase httpContext, ApplicationContext applicationContext)
+        // register current server (throttled).
+        private static void RegisterServer(UmbracoRequestEventArgs e)
         {
+            var reg = (DatabaseServerRegistrar) ServerRegistrarResolver.Current.Registrar;
+            var options = reg.Options;
+            var secondsSinceLastUpdate = DateTime.Now.Subtract(_lastUpdated).TotalSeconds;
+            if (secondsSinceLastUpdate < options.ThrottleSeconds) return;
+
+            _lastUpdated = DateTime.Now;
+
+            var url = e.HttpContext.Request.Url;
+            var svc = e.UmbracoContext.Application.Services.ServerRegistrationService;
+
             try
             {
-                var address = httpContext.Request.Url.GetLeftPart(UriPartial.Authority);
-                applicationContext.Services.ServerRegistrationService.EnsureActive(address);
+                if (url == null)
+                    throw new Exception("Request.Url is null.");
+
+                var serverAddress = url.GetLeftPart(UriPartial.Authority);
+                var serverIdentity = JsonConvert.SerializeObject(new

```

Comment:
I'm going to create a ctor overload for DatabaseServerRegistrar that accepts a method for which people can define that will be able to generate a custom server identity, since if we do end up using this for a real thing in the future, a server identity for Azure websites vs a normal server will be vastly different so we need to make this extensible.